File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int i;
        for( i=0;i<nums.length;i++){
            if(target<nums[i]){
                return i;
            }
            if(nums[i]==target){
                return i;
            }
            else if(i!=0 &&nums[i-1]<target && nums[i]>target){
                return i;
            }
        }
        return i;
}
}

File: 35-search-insert-position/search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        l=0
        r=len(nums)-1
        while l<=r:
            m=(l+r)/2
            if nums[m]==target:
                return m
            elif nums[m]<target:
                l+=1
            else:
                r-=1
        if nums[m]<target:
            return m+1
        return m